source/common/grpc/grpc_web_filter.cc

```diff
@@ -0,0 +1,109 @@
+#include "common/grpc/grpc_web_filter.h"
+
+#include <arpa/inet.h>
+
+#include "common/common/base64.h"
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Grpc {
+
+const uint8_t GrpcWebFilter::GRPC_WEB_TRAILER = 0b10000000;
+
+GrpcWebFilter::GrpcWebFilter() : is_text_request_(false), is_text_response_(false) {}
+
+GrpcWebFilter::~GrpcWebFilter() {}
+
+// Implements StreamDecoderFilter.
+Http::FilterHeadersStatus GrpcWebFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  if (content_type != nullptr &&
+      Http::Headers::get().ContentTypeValues.GrpcWebText == content_type->value().c_str()) {
```

Comment:
Based on https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2, I would expect that grpc-web-text to be a prefix, not exact match. The default is "+proto", but what about the other cases?